nengo/params.py

```diff
@@ -416,6 +426,47 @@ def validate(self, instance, function_info):
         super(FunctionParam, self).validate(instance, function)
 
 
+class CopyableObject(object):
```

Comment:
I'm not sure if this should be a mixin class. We're just using it in one place right now. If we just build this stuff right into `NengoObject`, then we don't have to do the `hasattr` checks for `__getstate__` and `__setstate__` below, and I think it's clearer.

This may at some point be reuseable functionality that we want to abstract out like this. But until we find a potential second use, I'd just build this into `NengoObject`.